CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+  - Stratum V2 [Master Degree Thesis](https://github.com/GitGab19/Stratum-V2-Master-Degree-Thesis) (by [@gitgab19](https://github.com/GitGab19/)).
+    - This resource can be useful to get some knowledge about Bitcoin mining, pooled mining protocols history, and Stratum V2.  
+  - Stratum V2 Explained - [Videos Playlist](https://www.youtube.com/playlist?list=PLZXAi8dsUIn0GmElOcmqUtgA5psfFIZoO) (by [@plebhash](https://github.com/plebhash)). 
+    - This is a series of videos explaining Stratum V2 in depth, which cover the aforementioned topics.
+
+### Project Communications
+
+Most project communications happen in our [Discord](https://discord.gg/fsEW23wFYs) server. Communications related to general development typically happen under [dev](https://discord.com/channels/950687892169195530/958814900770205739) channel.
+
+Discussion about specific codebase work happens in GitHub [issues](https://github.com/stratum-mining/stratum/issues/) and on [pull requests](https://github.com/stratum-mining/stratum/pulls/).
+
+Our dev calls are scheduled every Tuesday at 18.00 CET. You can see them in the sidebar under Events on Discord and subscribe to them to be notified.
+
+## I Want To Contribute
+> When contributing to this project, you must agree that you have authored 100% of the content, that you have the necessary rights to the content and that the content you contribute may be provided under the project license.
+
+### Project Structure
+It's possible to contribute to SRI opening PRs on three different repositories:
+  - [Stratum V2 Reference Implementation](https://github.com/stratum-mining/stratum)
+    - This repo contains our implementation of the SV2 specs, written in Rust.
+  - [Stratum V2 Specifications](https://github.com/stratum-mining/sv2-specs)
+    - This repo contains the entire SV2 protocol specifications.
+  - [SRI website - stratumprotocol.org](https://github.com/stratum-mining/stratumprotocol.org)
+    - This repo manages our website, containing docs, specs, and getting-started guides.
+
+### Contribution workflow
+
+The SRI project follows an open contributor model, where anyone is welcome to contribute through reviews, documentation, testing, and patches. Follow these steps to contribute:
+
+1. **Fork the Repository**
+
+2. **Create a Branch** 
+
+3. **Commit Your Changes** 
+
+4. **Submit a Pull Request:** once you're satisfied with your changes, submit a pull request to the original SRI repository. Provide a clear and concise description of the changes you've made. If your pull request addresses an existing issue, reference the issue number in the description. In order to contribute to the protocol implementation, every PR must be opened against `dev` branch. To better understand which is the SRI branches structure, please have a look [here](https://github.com/stratum-mining/stratum/blob/main/RELEASE.md#principal-branches).
```

Comment:
maybe we should add something about running tests + clippy + fmt before open the PR. We could also adapt the little thing that I wrote about debugging MG tests here

Reply:
Done here 5544830909ab3d489db23815d9c5e91d6574b5e1